File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    // int beautySum(string s) {
    //     int ans=0;
    //     int n=s.length();
    //     for(int i=0;i<n;i++){
    //         unordered_map<char,int> mpp;
    //         multiset<int> st;
    //         for(int j=i;j<n;j++){
    //             if(mpp.find(s[j])!=mpp.end()){
    //                 st.erase(st.find(mpp[s[j]]));
    //             }
    //             mpp[s[j]]++;
    //             st.insert(mpp[s[j]]);
    //             ans+=(*st.rbegin()-*st.begin());
    //         }
    //     }
    //     return ans;
    // }
    int beauty(vector<int>& alpha) {
        int lf = 1e9;
        int mf = -1;
        for (int i = 0; i < alpha.size(); ++i) {
            mf = max(mf, alpha[i]);
            if (alpha[i] >= 1) {
                lf = min(lf, alpha[i]);
            }
        }
        return mf - lf;
    }

    int beautySum(string s) {
        int res = 0;
        for (int i = 0; i < s.length(); ++i) {
            vector<int> alpha(26, 0);
            for (int j = i; j < s.length(); ++j) {
                alpha[s[j] - 'a'] += 1;
                res += beauty(alpha);
            }
        }
        return res;
    }
};

File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.py
from collections import defaultdict

class Solution:
    def beauty(self, alpha: list) -> int:
        lf = float('-inf')
        mf = -1
        for i in range(len(alpha)):
            mf = max(mf, alpha[i])
            if alpha[i] >= 1:
                lf = min(lf, alpha[i])
        return mf - lf

    def beautySum(self, s: str) -> int:
        # res = 0
        # for i in range(len(s)):
        #     alpha = [0] * 26
        #     for j in range(i, len(s)):
        #         alpha[ord(s[j]) - ord('a')] += 1
        #         res = res + self.beauty(alpha)
        # return res
        ans = 0 
        for i in range(len(s)):
            freq = [0]*26
            for j in range(i, len(s)):
                freq[ord(s[j])-97] += 1
                ans += max(freq) - min(x for x in freq if x)
        return ans 